proto/nym/vpn.proto

```diff
@@ -353,78 +354,55 @@ message AsExit {
   bool can_route_ip_external_v6 = 5;
 }
 
+message WgProbeResult {
+  bool can_register = 1;
+  bool can_handshake = 2;
+  bool can_resolve_dns = 3;
+  float ping_hosts_performance = 4;
+  float ping_ips_performance = 5;
+}
+
 message ProbeOutcome {
   AsEntry as_entry = 1;
   AsExit as_exit = 2;
+  WgProbeResult wg = 3;
 }
 
 message Probe {
   google.protobuf.Timestamp last_updated_utc = 1;
   ProbeOutcome outcome = 2;
 }
 
-message EntryGateway {
-  Gateway id = 1;
-  Location location = 2;
-  Probe last_probe = 3;
-}
-
-message ListEntryGatewaysRequest {
-  Threshold min_gateway_performance = 1;
-}
-
-message ListEntryGatewaysResponse {
-  repeated EntryGateway gateways = 1;
-}
-
-message ExitGateway {
+message GatewayResponse {
   Gateway id = 1;
   Location location = 2;
   Probe last_probe = 3;
 }
 
-message ListExitGatewaysRequest {
-  Threshold min_gateway_performance = 1;
+enum GatewayType {
+  GATEWAY_TYPE_UNSPECIFIED = 0;
+  ENTRY = 1;
+  EXIT = 2;
+  VPN = 3;
 }
 
-message ListExitGatewaysResponse {
-  repeated ExitGateway gateways = 1;
+message ListGatewaysRequest {
+  GatewayType kind = 1;
+  Threshold min_mixnet_performance = 2;
+  Threshold min_vpn_performance = 3;
 }
 
-message VpnGateway {
-  Gateway id = 1;
-  Location location = 2;
-}
-
-message ListVpnGatewaysRequest {
-  Threshold min_gateway_performance = 1;
-}
-
-message ListVpnGatewaysResponse {
-  repeated VpnGateway gateways = 1;
-}
-
-message ListEntryCountriesRequest {
-  Threshold min_gateway_performance = 1;
-}
-
-message ListEntryCountriesResponse {
-  repeated Location countries = 1;
-}
-
-message ListExitCountriesRequest {
-  Threshold min_gateway_performance = 1;
-}
-
-message ListExitCountriesResponse {
-  repeated Location countries = 1;
+message ListGatewaysResponse {
+  repeated GatewayResponse gateways = 1;
 }
 
-message ListVpnCountriesRequest {
-  Threshold min_gateway_performance = 1;
+message ListCountriesRequest {
+  GatewayType kind = 1;
+  Threshold min_mixnet_performance = 2;
+  Threshold min_vpn_performance = 3;
```

Comment:
These two fields should be both marked as `optional`

Reply:
It is an `Option` in rust implementations yes, but is it the case with other languages' implementation, like in swift?

Reply:
@rokas-ambrazevicius are proto non-scalar types generated in Swift code as "optional" or "nullable" without being labeled as `optional`?